### PR TITLE
Constrain Augustus version

### PR DESCRIPTION
**Pull request by berkelem**
Update recipe to constrain the dependency Augustus to run only version 3.2.3.
----

> Bioconda requires reviews prior to merging pull-requests (PRs). To facilitate this, once your PR is passing tests and ready to be merged, please add the `please review & merge` label so other members of the bioconda community can have a look at your PR and either make suggestions or merge it.

* [x] I have read the [guidelines for bioconda recipes](https://bioconda.github.io/contributor/guidelines.html).
* [ ] This PR adds a new recipe.
* [x] AFAIK, this recipe **is directly relevant to the biological sciences**
      (otherwise, please submit to the more general purpose [conda-forge channel](https://conda-forge.org/docs/)).
* [x] This PR updates an existing recipe.
* [ ] This PR does something else (explain below).

> Everyone has access to the following BiocondaBot commands, which can be given in a comment:
>
>  * `@BiocondaBot please update` will cause the BiocondaBot to merge the master branch into a PR
>  * `@BiocondaBot please add label` will add the `please review & merge` label.
>  * `@BiocondaBot please fetch artifacts` will post links to packages and docker containers built by the CI system. You can use this to test packages locally before merging.
>
> For members of the Bioconda project, the following command is also available:
>
>  * `@BiocondaBot please merge` will cause packages/containers to be uploaded and a PR merged. Someone must approve a PR first! This has the benefit of not wasting CI build time required by manually merging PRs.
>
> If you have questions, please post them in gitter or ping `@bioconda/core` in a comment (if you are not able to directly ping `@bioconda/core` then the bot will repost your comment and enable pinging).

### CODE REVIEW

**Comment by berkelem** [comment]
@BiocondaBot please add label

File: recipes/busco/meta.yaml [hunk 1]
```diff
@@ -24,7 +24,7 @@ requirements:
     - findutils
     - blast >=2.2,<2.4
     - hmmer >=3.1b2 # sepp needs this version
-    - augustus
+    - augustus 3.2.3
```
Comment (SilasK) [inline recipes/busco/meta.yaml]: Do you need exactly this version. I suggest to use `augustus >= 3.2.3`

Comment (berkelem) [reply recipes/busco/meta.yaml]: For now, yes we just want 3.2.3. There are discrepancies in the results when using 3.3.x that we are trying to get to the bottom of, so we don't want >= 3.2.3.

**Comment by SilasK** [comment]
@BiocondaBot please merge

**Pull request merged**